packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -291,6 +330,10 @@ export default util.createRule<Options, MessageIds>({
               modifiers.add(Modifiers.unused);
             }
 
+            if (node.async) {
+              modifiers.add(Modifiers.async);
```

Comment:
Hmm, codecov is right - this isn't covered by tests. Could you please add a valid & invalid pair for this case? (unless the case is unreachable, in which case it can be removed)

Reply:
Yeah good catch by codecov, I think this was a mistake as I don't think its possible 😅 since `async` can only be used directly on a function/arrow function expression not identifiers

Removed